### PR TITLE
Avoid TypeError when referenced attribute is undefined

### PR DESCRIPTION
**Pull request by csarven**
This should avoid unnecessary TypeErrors when `rel`, `get`, `has` is not guaranteed to find matching attribute before attempting to operate on the string value.

### CODE REVIEW

File: lib/link.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ class Link {
     var type = value.toLowerCase()
 
     for( var i = 0; i < this.refs.length; i++ ) {
-      if( this.refs[ i ].rel.toLowerCase() === type ) {
+      if( this.refs[ i ].rel && this.refs[ i ].rel.toLowerCase() === type ) {
```
Comment (jhermsmeier) [inline lib/link.js:88]: Should this null- or typeof-check instead of relying on implicit falsy-ness? Thinking an empty `rel` could trip this up, e.g. `<example.com>; rel=`. Not sure if that's a real use-case, and don't remember if the specs would allow it though.

```suggestion
      if( typeof this.refs[ i ].rel == 'string' && this.refs[ i ].rel.toLowerCase() === type ) {
```

Comment (jhermsmeier) [reply lib/link.js:88]: Would also be a good opportunity to add test cases for `rel()`, `get()`, and `has()` with empty `rel`s in `test/api`, but I can also do that after merging this, so more of a note-to-self I guess :)

Comment (csarven) [reply lib/link.js:88]: I thought of typeof as well, and we can go with that. Though, isn't the lowerCase of empty string the same empty string? `rel` with empty value is not conforming as per RFC8288 but not sure if it matters here either.

File: lib/link.js [hunk 2]
```diff
@@ -108,7 +108,7 @@ class Link {
     var links = []
 
     for( var i = 0; i < this.refs.length; i++ ) {
-      if( this.refs[ i ][ attr ].toLowerCase() === value ) {
+      if( this.refs[ i ][ attr ] && this.refs[ i ][ attr ].toLowerCase() === value ) {
```
Comment (csarven) [inline lib/link.js:111]:
```suggestion
      if( typeof this.refs[ i ][ attr ] === 'string' && this.refs[ i ][ attr ].toLowerCase() === value ) {
```

File: lib/link.js [hunk 3]
```diff
@@ -142,7 +142,7 @@ class Link {
     value = value.toLowerCase()
 
     for( var i = 0; i < this.refs.length; i++ ) {
-      if( this.refs[ i ][ attr ].toLowerCase() === value ) {
+      if( this.refs[ i ][ attr ] && this.refs[ i ][ attr ].toLowerCase() === value ) {
```
Comment (csarven) [inline lib/link.js:145]:
```suggestion
      if( typeof this.refs[ i ][ attr ] === 'string' && this.refs[ i ][ attr ].toLowerCase() === value ) {
```

File: lib/link.js (hunk 1 above)
Comment (csarven) [reply lib/link.js:88]: Perhaps change '===' instead of '=='. I've used the same approach for the others.

Comment (jhermsmeier) [reply lib/link.js:88]: > Though, isn't the lowerCase of empty string the same empty string? 

Yes, but it wouldn't get to the `rel.toLowerCase() === type` comparison, because for `rel = ''`, the `this.refs[ i ].rel &&` would be evaluate to `false`, which in turn would not yield the expected result; e.g. `.has( 'rel', '' )` would always return `false`, even if an entry with an empty `rel` would be present.

Was thinking about cases where this module is used to validate headers, where one might want to check for things like this.

Comment (jhermsmeier) [reply lib/link.js:88]: Got confused with the impl's of `.rel()` and `.has()` here, but same goes for `.rel( '' )`

**Comment by jhermsmeier** [comment]
Thanks, again! Published in `http-link-header@1.1.3`
